Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+                    {
+                        //TODO Native plugins aren't yet supported
+                        continue;
+                    }
+                    List<string> platformConditions = GetPlatformConditions(inEditor ? InEditorPlatforms : PlayerPlatforms, inEditor ? dependency.InEditorSupportedPlatforms : dependency.PlayerSupportedPlatforms);
+
+                    pluginReferences.Add(Utilities.ReplaceTokens(pluginReferenceTemplate, new Dictionary<string, string>()
+                    {
+                        { "##REFERENCE_TOKEN##", dependency.Dependency.Name },
+                        { "##CONDITION_TOKEN##", platformConditions.Count == 0 ? "'false'" : string.Join(" OR ", platformConditions)},
+                        { "<!--HINT_PATH_TOKEN-->", dependency.Dependency.ReferencePath.AbsolutePath }
+                    }));
+
+                    additionalSearchPaths.Add(Path.GetDirectoryName(dependency.Dependency.ReferencePath.AbsolutePath));
+                }
+
+                return Utilities.ReplaceTokens(template, new Dictionary<string, string>()
+                {
+                    {"##REFERENCE_CONFIGURATION_TOKEN##", inEditor ? "InEditor" : "Player" },
+                    { projectReferenceTemplate, string.Join("\r\n", projectReferences) },
+                    { pluginReferenceTemplate, string.Join("\r\n", pluginReferences) }
+                });
+            }
+            else
+            {
+                Debug.LogError("Failed to find ProjectReference template in ProjectReferenceSet template");
+                return template;
+            }
+        }
+
+        private List<string> GetPlatformConditions(IReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> platforms, HashSet<BuildTarget> dependencyPlatforms)
+        {
+            List<string> toReturn = new List<string>();
+
+            foreach (KeyValuePair<BuildTarget, CompilationPlatformInfo> pair in platforms)
+            {
+                if (dependencyPlatforms.Contains(pair.Key))
+                {
+                    string platformName = pair.Value.Name;
+                    toReturn.Add($"'$(UnityPlatform)' == '{platformName}'");
+                }
+            }
+
+            return toReturn;
+        }
+
+        private string GetDefaultPlatform(IReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> platforms)
+        {
+
+            if (platforms.TryGetValue(BuildTarget.StandaloneWindows, out CompilationPlatformInfo defaultPlatform)
+                || platforms.TryGetValue(BuildTarget.StandaloneWindows64, out defaultPlatform)
+                || platforms.TryGetValue(BuildTarget.WSAPlayer, out defaultPlatform)
+                || platforms.TryGetValue(BuildTarget.Android, out defaultPlatform)
+                || platforms.TryGetValue(BuildTarget.iOS, out defaultPlatform)
+                || platforms.TryGetValue(BuildTarget.NoTarget, out defaultPlatform))
```

Comment:
Should this be kept in sync with the define:

supportedBuildTargets

In MSBuildTools.cs?

Reply:
Yeah, but no longer used :)

---
In reply to: [301802009](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301802009) [](ancestors = 301802009)